pyscf/nao/gw_iter.py

```diff
@@ -290,34 +290,36 @@ def gw_chi0_mv(self,dvin, comega=1j*0.0, dnout=None):
     from scipy.linalg import blas
     from pyscf.nao.m_sparsetools import csr_matvec    
     
-    if dnout is None:
-        dnout = np.zeros_like(dvin, dtype=self.dtypeComplex)
+    # real part
+    vdp = csr_matvec(self.cc_da_csr, dvin.real)
+    #sab_real = (vdp*self.v_dab).reshape((self.norbs,self.norbs))
+    sab_real = csr_matvec(self.v_dab_trans, vdp).reshape((self.norbs,self.norbs))
 
-    dnout.fill(0.0)
-
-    vdp = csr_matvec(self.cc_da, dvin.real)  # real part
-    sab_real = (vdp*self.v_dab).reshape((self.norbs,self.norbs))
-
-    vdp = csr_matvec(self.cc_da, dvin.imag)  # imaginary
-    sab_imag = (vdp*self.v_dab).reshape((self.norbs, self.norbs))
+    # imaginary
+    vdp = csr_matvec(self.cc_da_csr, dvin.imag)
+    #sab_imag = (vdp*self.v_dab).reshape((self.norbs, self.norbs))
+    sab_imag = csr_matvec(self.v_dab_trans, vdp).reshape((self.norbs, self.norbs))
 
     ab2v_re = np.zeros((self.norbs, self.norbs), dtype=self.dtype)
     ab2v_im = np.zeros((self.norbs, self.norbs), dtype=self.dtype)
 
-    for s in range(self.nspin):
-        nb2v = self.gemm(1.0, self.xocc[s], sab_real)
-        nm2v_re = self.gemm(1.0, nb2v, self.xvrt[s].T)
+    for spin in range(self.nspin):
```

Comment:
+1 for s-->spin

Reply:
I really believe that there is an issue with this loop, the results are not saved in an array. Therefore, only the last spin element will count. Could you have a look if you have some time?

I will open a new merge request with what I think should be the proper way.

Reply:
There must be accumulation in ab2v_re (and ab2v_im) in this loop over spin.